File: src/com/huangyueran/spark/sql/SpecifyFormatLoadSave.java
package com.huangyueran.spark.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.DataFrameReader;
import org.apache.spark.sql.DataFrameWriter;
import org.apache.spark.sql.SQLContext;

/**
 * @category format()简单格式的格式转换存储
 * @author huangyueran
 * @time 2017-7-24 14:33:23
 */
public class SpecifyFormatLoadSave {
	public static void main(String[] args) {
		SparkConf conf = new SparkConf().setAppName("SpecifyFormatLoadSave").setMaster("local");
		JavaSparkContext sc = new JavaSparkContext(conf);

		// 创建DataFrame 读取json
		SQLContext sqlContext = new SQLContext(sc);

		DataFrameReader dataFrameReader = sqlContext.read();

		// parquet 是本地数据存储的格式
		DataFrame df = dataFrameReader.format("json").load("resources/people.json");

		// 通过关writer写入并保存save
		DataFrameWriter write = df.select("name").write();
		write.format("parquet").save("people.parquet");

		sc.close();
	}
}

File: src/main/scala/com/hyr/spark/sql/SpecifyFormatLoadSave.scala
package com.hyr.spark.sql

import com.huangyueran.spark.utils.Constant
import com.hyr.spark.utils.SparkUtils
import org.apache.spark.sql.SparkSession

/** *****************************************************************************
  * @date 2019-08-30 09:42
  * @author: <a href=mailto:huangyr>黄跃然</a>
  * @Description:
  * *****************************************************************************/
object SpecifyFormatLoadSave {

  def main(args: Array[String]): Unit = {
    val sparkContext = SparkUtils.getRemoteSparkContext(SpecifyFormatLoadSave.getClass)
    val sparkSession = SparkSession.builder().appName("SpecifyFormatLoadSave").getOrCreate()

    val dataFrameReader = sparkSession.read
    val dataset = dataFrameReader.format("json").load( "/data/resources/people.json")

    // 通过关writer写入并保存save
    val write = dataset.select("name").write
    write.format("parquet").save("tmp/people.parquet")

  }

}
